<?xml version="1.0"?>
<!-- Note: example launch file. Do not modify! -->
<launch>
  <arg name="loop_rate" default="500"/>
  <arg name="sequence_length" default="20"/>
  <arg name="delay_predicted" default="1"/>
  <arg name="delay_actual" default="80"/>
  <arg name="filter_samples" default="300"/>
  <arg name="model_name" default="linear_accel_orientation_twist"/>
  <arg name="node_name" default="force_sensor_model"/>
  <arg name="use_remote_machine" default="false"/>
  <arg name="machine" default="None"/>

  <group if="$(arg use_remote_machine)">
    <node pkg="udom_sensor_model" type="force_sensor_model"
          name="$(arg node_name)" output="screen" machine="$(arg machine)">
      <!-- Number of samples representing the sequence length that was used to learn the model.-->
      <param name="sequence_length" type="int" value="$(arg sequence_length)"/>

      <!-- Number of samples to delay the actual wrench such that it matches the delayed model's estimation. -->
      <param name="delay_predicted" type="double" value="$(arg delay_predicted)"/>

      <!-- Number of samples to delay the estimated wrench such that it matches the actual wrench. -->
      <param name="delay_actual" type="double" value="$(arg delay_actual)"/>

      <!-- Number of samples to filter the estimated non-contact wrench.-->
      <param name="filter_samples" type="double" value="$(arg filter_samples)"/>

      <!-- Name of the model for the trained RNN. -->
      <param name="model_name" type="str" value="$(arg model_name)"/>

      <!-- Node cycle rate (in Hz). -->
      <param name="loop_rate" type="double" value="$(arg loop_rate)"/>
    </node>
  </group>

  <group unless="$(arg use_remote_machine)">
    <node pkg="udom_sensor_model" type="force_sensor_model"
          name="$(arg node_name)" output="screen">
      <!-- Number of samples representing the sequence length that was used to learn the model.-->
      <param name="sequence_length" type="int" value="$(arg sequence_length)"/>

      <!-- Number of samples to delay the actual wrench such that it matches the delayed model's estimation. -->
      <param name="delay_predicted" type="double" value="$(arg delay_predicted)"/>

      <!-- Number of samples to delay the estimated wrench such that it matches the actual wrench. -->
      <param name="delay_actual" type="double" value="$(arg delay_actual)"/>

      <!-- Number of samples to filter the estimated non-contact wrench.-->
      <param name="filter_samples" type="double" value="$(arg filter_samples)"/>

      <!-- Name of the model for the trained RNN. -->
      <param name="model_name" type="str" value="$(arg model_name)"/>

      <!-- Node cycle rate (in Hz). -->
      <param name="loop_rate" type="double" value="$(arg loop_rate)"/>
    </node>
  </group>
</launch>
